File: src/sp/it/pl/util/reactive/SetƑ.java
package sp.it.pl.util.reactive;

import java.util.HashSet;
import org.reactfx.Subscription;
import sp.it.pl.util.functional.Functors.Ƒ;

/** Set of {@link java.lang.Runnable} taking o parameters. Use as a collection of handlers. */
public class SetƑ extends HashSet<Runnable> implements Ƒ {

	public SetƑ() {
		super(2);
	}

	@Override
	public void apply() {
		forEach(Runnable::run);
	}

	public Subscription addS(Runnable r) {
		add(r);
		return () -> remove(r);
	}

}

File: src/sp/it/pl/util/reactive/SetƑ.kt
package sp.it.pl.util.reactive

import org.reactfx.Subscription
import sp.it.pl.util.functional.Functors.Ƒ
import sp.it.pl.util.functional.invoke
import java.util.HashSet

/** Set of actions taking 0 parameters. Use as a collection of handlers. */
class SetƑ: HashSet<Runnable>(2), Ƒ {

    override fun apply() {
        forEach { it() }
    }

    fun addS(r: Runnable): Subscription {
        add(r)
        return Subscription { remove(r) }
    }

    fun addS(r: () -> Unit) = addS(Runnable { r() })

}